app/Http/Resources/Topic.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace App\Http\Resources;
+
+use Illuminate\Http\Resources\Json\JsonResource;
+
+class Topic extends JsonResource
+{
+    /**
+     * Transform the resource into an array.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return array
+     */
+    public function toArray($request)
+    {
+        return [
+            'id' => $this->id,
+            'name' => $this->name,
+            'description' => $this->description,
+            'subject_id' => $this->subject_id,
```

Comment:
```suggestion
            'subject' => SubjectShortResource::make($this->subject),
```

Reply:
$this->subject is a BelongsTo relation on the Subject model.

Also, make a SubjectShortResource. It should contain only the id and name of the subject.